### PR TITLE
Consolidate make_tt_tpl and make_tt

### PR DESCRIPTION
**Pull request by devreal**
We can map `make_tt_tpl` onto the same mechanism we use for `make_tt` by adding a template template parameter that signals whether arguments are passed as tuple or unpacked.

This is an attempt to clean up the implementation and to open the door for device-support in `ttg::make_tt_tpl`.

### CODE REVIEW

File: ttg/ttg/util/finally.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+//
+///////////////////////////////////////////////////////////////////////////////
+
+namespace ttg::detail {
+
+    /* finally, taken from GSL: https://github.com/microsoft/GSL */
+
+    // final_action allows you to ensure something gets run at the end of a scope
+    template <class F>
+    class final_action
```
Comment (evaleev) [inline ttg/ttg/util/finally.h:26]: we might want to consider something more fully featured like https://github.com/justusc/deferral

Comment (evaleev) [reply ttg/ttg/util/finally.h:26]: at least let's rename to match what will eventually land in C++: https://en.cppreference.com/w/cpp/experimental/scope_exit/scope_exit

Comment (devreal) [reply ttg/ttg/util/finally.h]: I took the implementation from https://github.com/tandasat/ScopedResource/ and slightly adopted it.

**Comment by devreal** [comment]
@evaleev I replace `finally` with `exit_scope`. I think this is ready to go.

**Pull request merged**